controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -232,9 +239,15 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 				}
 			}
 		default:
-			err := r.createUserGroup(ctx, instance, "odh-admins")
-			if err != nil {
-				return reconcile.Result{}, err
+			// Check if user opted for disabling creating user groups
+			disableUserGroup, exist := os.LookupEnv("DISABLE_USERGROUP")
+			if exist && disableUserGroup != "false" {
```

Comment:
nit: since we don't need to distinguish between between an empty value and an unset value, the check can probably simplified with `os.Getenv("DISABLE_USERGROUP") == "true"`